src/loader/mod.rs

```diff
@@ -75,22 +75,21 @@ pub type Result<T> = std::result::Result<T, Error>;
 
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let desc = match self {
+        write!(f, "Kernel Loader: ")?;
+        match self {
             #[cfg(all(feature = "bzimage", any(target_arch = "x86", target_arch = "x86_64")))]
-            Error::Bzimage(ref _e) => "failed to load bzImage kernel image",
+            Error::Bzimage(ref e) => write!(f, "failed to load bzImage kernel image: {e}"),
```

Comment:
```suggestion
            Error::Bzimage(ref e) => format!("failed to load bzImage kernel image: {e}"),
```

Reply:
How about use `format!` in the `match` block, and call `write!` once to print the inner message. Would calling `write!` twice lead to performance regression :)

Reply:
The issue with `format!` is that we will need to make temporary strings (e.g. do allocations) for all error cases. I can add the `Kernel Loader: ` part of the first `write` into all writes so there will be only 1 `write!` call. But in any case, this is an error case, so I doubt the performance is a big concern here.

Reply:
Silly of me, I recollected these "write"s previously when dealing with write operation into `fifo`s, and indeed this is in an error case.

LGTM :)